### PR TITLE
SSR-735-list app widget

### PR DESCRIPTION
**Pull request by Pham-Gia-Huong**

### CODE REVIEW

File: src/renderer/Layout/GridLayout/GridBlock/index.tsx [hunk 1]
```diff
@@ -169,6 +170,26 @@ const GridBlock = ({ style, content = undefined, width, rowIndex, blockIndex, on
...
+              onSaveSetting={({listContent, titleWidget}) => {
+                let currentProps = JSON.parse(JSON.stringify(currentBlock.content));
+                currentProps.listContent = listContent;
+                currentProps.titleWidget = titleWidget;
+                currentProps.showSettingInit = false;
+                updateWidget(currentProps);
+              }}
+            />
+          );
+          break;
```
Comment (phongnhat19) [inline src/renderer/Layout/GridLayout/GridBlock/index.tsx]: Re-run eslint on this code block

File: src/renderer/Layout/GridLayout/GridBlock/index.tsx [hunk 2]
```diff
@@ -225,6 +246,10 @@ const GridBlock = ({ style, content = undefined, width, rowIndex, blockIndex, on
             openWeatherMapAPIKey: '',
             type: WEATHER_TYPE.SIMPLE
           }
+        } else if (type === CONTENT_TYPE.APP_SPACE) {
+          props = {
+            showSettingInit: true,
+          };
```
Comment (phongnhat19) [inline src/renderer/Layout/GridLayout/GridBlock/index.tsx:294]: Re-run eslint on this code block

File: src/renderer/Layout/GridLayout/GridBlock/renderer.tsx [hunk 3]
```diff
@@ -70,6 +71,12 @@ const GridBlock = ({ style, type = CONTENT_TYPE.EMPTY as ContentType, content =
               type={blockContentWeather.type}
             />
           break;
+          case CONTENT_TYPE.APP_SPACE:
+            const blockContentAppSpace = content as AppSpaceWidgetProps;
+            currentContentBlock = (
+              <AppSpaceWidget titleWidget={blockContentAppSpace.titleWidget} listContent={blockContentAppSpace.listContent} />
+            );
+            break;
```
Comment (phongnhat19) [inline src/renderer/Layout/GridLayout/GridBlock/renderer.tsx]: Re-run eslint on this code block

File: src/renderer/Layout/TabsLayout/index.tsx [hunk 4]
```diff
@@ -128,6 +129,25 @@ const TabsLayout = ({
...
+              onSaveSetting={({listContent, titleWidget}) => {
+                let currentProps = JSON.parse(JSON.stringify(tabContent.props));
+                currentProps.listContent = listContent;
+                currentProps.titleWidget = titleWidget;
+                currentProps.showSettingInit = false;
+                updateWidget(currentProps);
+              }}
+            />
+          );
+          break;
```
Comment (phongnhat19) [inline src/renderer/Layout/TabsLayout/index.tsx]: Re-run eslint on this code block

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
+import React, {useState} from 'react';
+import {Modal} from 'antd';
+import {PlusCircleFilled, MinusCircleFilled} from '@ant-design/icons';
+import {InputText} from './components/InputWidget';
+import './style.css';
+import RenderCategoryDetail from './components/categoryList';
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Module name is `CategoryList` not `categoryList`

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+          value={content.category}
+          placeholder={INPUT_TEXT.CATEGORY}
+          type={'text'}
+          className={''}
+          onChange={(value) => {
+            let valueString = value as string;
+            newData.category = valueString;
+            onChange(newData, '');
+          }}
+        />
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  content,
+  onChange,
+  i,
+}: {
+  content: ModalAppSpace;
+  onChange: (content: ModalAppSpace, status: string) => void;
+  i: number;
+}) => {
+  let newData = {...content};
+  let listCategory = newData.listCategory as CategorytAppSpace[];
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+import {INPUT_TEXT, MODAL, LABEL} from './constant';
+
+const RenderBlockCategory = ({
+  content,
+  onChange,
+  i,
+}: {
+  content: ModalAppSpace;
+  onChange: (content: ModalAppSpace, status: string) => void;
+  i: number;
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Rename props for clear meaning

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 9]
```diff
@@ -0,0 +1,167 @@
+import React, {useState} from 'react';
+import {Modal} from 'antd';
+import {PlusCircleFilled, MinusCircleFilled} from '@ant-design/icons';
+import {InputText} from './components/InputWidget';
+import './style.css';
+import RenderCategoryDetail from './components/categoryList';
+
+import {INPUT_TEXT, MODAL, LABEL} from './constant';
+
+const RenderBlockCategory = ({
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Component name should be a noun, not a verb

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+        onChangeRow={(newListCategory) => {
+          newData.listCategory = newListCategory;
+          onChange(newData, '');
+        }}
+      />
+    </div>
+  );
+};
+
+const RenderCategory = ({
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Component name should be a noun, not a verb

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 11]
```diff
@@ -0,0 +1,167 @@
...
+              i={i}
+            />
+          </React.Fragment>
+        );
+      })}
+    </div>
+  );
+};
+
+const RenderActionAddCategory = ({
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Component name should be a noun, not a verb

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+          }}
+          i={i}
+          onClick={(value, status) => {
+            onChange(value, status);
+          }}
+        />
+      </div>
+
+      <RenderCategoryDetail
+        listCategory={listCategory}
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Should be `categoryList`, not `listCategory`

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+          }}
+        />
+        <RenderActionAddCategory
+          onRemove={(status) => {
+            onChange(newData, status);
+          }}
+          i={i}
+          onClick={(value, status) => {
+            onChange(value, status);
+          }}
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Could be change to `onClick={onChange}` for shorter code

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+                }
+                setListContent(newListContent);
+              }}
+              i={i}
+            />
+          </React.Fragment>
+        );
+      })}
+    </div>
+  );
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+};
+
+const RenderActionAddCategory = ({
+  onClick,
+  onRemove,
+  i,
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Rename props for clear meaning

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+                } else if (status === 'remove') {
+                  newListContent.splice(i, 1);
+                } else {
+                  newListContent[i] = cloneData;
+                }
+                setListContent(newListContent);
+              }}
+              i={i}
+            />
+          </React.Fragment>
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: No need to wrap inside a `React.Fragment` component

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+  onClick,
+  onRemove,
+  i,
+}: {
+  onClick: (value: ModalAppSpace, status: string) => void;
+  onRemove: (status: string) => void;
+  i: number;
+}) => {
+  const handleAddCategory = () => {
+    let category: ModalAppSpace = {listCategory: [{type: 'app', id: '', name: '', icon: ''}], category: ''};
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Re-run eslint

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+    onClick(category, 'add');
+  };
+  const handleRemoveCategory = () => {
+    onRemove('remove');
+  };
+  return (
+    <div className="add-category">
+      <PlusCircleFilled onClick={() => handleAddCategory()} />
+      {i > 0 ? <MinusCircleFilled onClick={() => handleRemoveCategory()} /> : ''}
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Should be `{i > 0 && <MinusCircleFilled onClick={() => handleRemoveCategory()} />}`

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+      bodyStyle={{maxHeight: '500px', overflow: 'auto'}}
+      onOk={() => {
+        getContent({listContent: newListContent, titleWidget: newTitleWidget});
+        onCancel();
+      }}
+      title={MODAL.TITLE}
+      visible={showSettingInit}
+      onCancel={() => {
+        onCancel()
+      }}
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Should be `onCancel={onCancel}`

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+    >
+      <InputText
+        label={LABEL.TITLE}
+        width={'calc(94% - 5px)'}
+        placeholder={`${INPUT_TEXT.TITLE}`}
+        type={'text'}
+        value={newTitleWidget}
+        className="item-margin-bottom-10 title"
+        onChange={(value) => setTitleWidget(value as string)}
+      />
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: Re-run eslint

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+import {INPUT_TEXT} from '../constant';
+
+const RenderCategoryDetail = ({
+  listCategory,
+  onChangeRow,
+}: {
+  listCategory: any;
+  onChangeRow?: (listCategory: {id: string; type: string}[]) => void;
+}) => {
+  let newListCategory = listCategory.slice();
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx]: Re-run eslint

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+          type={'number'}
+          className="input-text-custome"
+          onChange={(value) => {
+            newListCategory[i].id = value as string;
+            if (onChangeRow) onChangeRow(newListCategory);
+          }}
+        />
+        <div className="item-block width">
+          <PlusCircleFilled onClick={() => addRow(i)} style={{color:"#1890ff"}} />
+          {i > 0 ? <MinusCircleFilled style={{color:"#1890ff"}} onClick={() => removeRow(i)} /> : ''}
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx]: Should be `{i > 0 && <MinusCircleFilled style={{color:"#1890ff"}} onClick={() => removeRow(i)} />}`

File: src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx [hunk 23]
```diff
@@ -0,0 +1,57 @@
...
+        <div className="item-block flex-end">
+          <DropDownType
+            value={newCategory.type}
+            onChange={(value: string) => {
+              newListCategory[i].type = value;
+              if (onChangeRow) onChangeRow(newListCategory);
+            }}
+          />
+        </div>
+        <InputText
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/CategoryList.tsx]: Rename component for clear meaning (for example `InputWithLabel`)

File: src/renderer/Widget/AppSpaceListWidget/components/DropDownType.tsx [hunk 24]
```diff
@@ -0,0 +1,27 @@
...
+        <span>App</span>
+      </Menu.Item>
+      <Menu.Item key="1">
+        <span>Space</span>
+      </Menu.Item>
+    </Menu>
+  );
+  return (
+    <Dropdown overlay={menu} trigger={['click']}>
+      <a className="ant-dropdown-link" onClick={(e) => e.preventDefault()}>
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/DropDownType.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/DropDownType.tsx [hunk 25]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import {Menu, Dropdown} from 'antd';
+import {DownOutlined} from '@ant-design/icons';
+
+const DropDownType = ({value, onChange}: {value: string; onChange: (value: string) => void}) => {
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/DropDownType.tsx]: Should be `TypeDropdown`, not `DropDownType`

File: src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+  width: string;
+  value: string;
+  placeholder: string;
+  type: string;
+  className:string;
+  disabled?: boolean;
+  onChange: (value: string | number) => void;
+}) => {
+  return (
+    <div className={`item-block ` + className}  style={{width}}>
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+  onChange,
+  className,
+  disabled
+}: {
+  label: string;
+  width: string;
+  value: string;
+  placeholder: string;
+  type: string;
+  className:string;
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx [hunk 28]
```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import {Input} from 'antd';
+
+export const InputText = ({
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx]: Component name should be consistent with file name

File: src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx (hunk 26 above)
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/InputWidget.tsx]: Should be ``className={`item-block ${className}`}``

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx [hunk 29]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+const IconType = ({type}: {type: string}) => {
+  let appType = type === 'app';
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: Re-run eslint here

Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: A boolean variable should be name as `isTypeApp`

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx [hunk 30]
```diff
@@ -0,0 +1,46 @@
...
+  return <span className={appType ? 'app-type border-radius' : 'app-type'}>{appType ? "Icon" : "Cover"}</span>;
+};
+
+const PreviewContent = ({listContent}: {listContent: ModalAppSpace[]}) => {
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: Should be `contentList`, not `listContent`

Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx [hunk 31]
```diff
@@ -0,0 +1,46 @@
...
+  return <span className={appType ? 'app-type border-radius' : 'app-type'}>{appType ? "Icon" : "Cover"}</span>;
+};
+
+const PreviewContent = ({listContent}: {listContent: ModalAppSpace[]}) => {  
+  const checkEvenNumber = (number: number) => {
+    if (number % 2 === 0) {
+      return true;
+    }
+    return false;
+  };
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx [hunk 32]
```diff
@@ -0,0 +1,46 @@
...
+  };  
+  return (
+    <div className="widget-content-preview">
+      {listContent.map((content,key) => {        
+        return (
+          <React.Fragment key={key}>
+            <div className="widget-content-preview-category">{content.category}</div>
+            {content.listCategory.map((category, i) => {
+              let numberEven = checkEvenNumber(i);
+              let hightLightRow= !numberEven ? "hight-light" : "";
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx [hunk 33]
```diff
@@ -0,0 +1,46 @@
...
+    return false;
+  };  
+  return (
+    <div className="widget-content-preview">
+      {listContent.map((content,key) => {        
+        return (
+          <React.Fragment key={key}>
+            <div className="widget-content-preview-category">{content.category}</div>
+            {content.listCategory.map((category, i) => {
+              let numberEven = checkEvenNumber(i);
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: Boolean variable should be named `isEven`

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx [hunk 34]
```diff
@@ -0,0 +1,46 @@
...
+                    <div className="widget-content-preview-col-icon">
+                      {category.icon ? (
+                        <img src={category.icon ? category.icon : ''} width={34} height={34} />
+                      ) : (
+                        <IconType type={category.type} />
+                      )}
+                    </div>
+                    <div className="widget-content-preview-col-app">{category.name ? category.name : `${category.type === "app" ? "App" : "Space"} name (id: ${category.id})`}</div>
+                  </div>
+                </React.Fragment>
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: No need to wrap inside `React.Fragment` component

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewTitle.tsx [hunk 35]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { AppstoreFilled } from '@ant-design/icons';
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewTitle.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewTitle.tsx [hunk 36]
```diff
@@ -0,0 +1,13 @@
...
+  return (
+    <div className="widget-header-preview">
+      <div className="widget-header-preview-icon">
+        <AppstoreFilled />
+      </div>
+      <div className="widget-header-preview-title">{titleWidget}</div>
+    </div>
+  );
+};
+export default PreviewTitle
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/components/PreviewTitle.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/index.tsx [hunk 37]
```diff
@@ -0,0 +1,39 @@
...
+  showSettingInit?: boolean;
+  onRemove?: () => void;
+  onSaveSetting?: ({listContent, titleWidget}: {listContent: any; titleWidget: string;}) => void;
+}) => {  
+  const [showSetting, setShowSetting] = useState(showSettingInit);
+  return (
+    <React.Fragment>
+      <SettingsWidget onRemove={onRemove} showSetting={() => setShowSetting(true)} />
+      <AppSpaceModel
+      titleWidget={titleWidget}
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/index.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/index.tsx [hunk 38]
```diff
@@ -0,0 +1,39 @@
...
+        listContent={listContent}
+        showSettingInit={showSetting}
+        getContent={({listContent, titleWidget}) => {
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/index.tsx]: `listContent` and `titleWidget` is re-declared

File: src/renderer/Widget/AppSpaceListWidget/index.tsx [hunk 39]
```diff
@@ -0,0 +1,39 @@
...
+          let newListContent = listContent.slice()
+          onSaveSetting && onSaveSetting({listContent:newListContent, titleWidget});
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/index.tsx]: Re-run eslint here

File: src/renderer/Widget/AppSpaceListWidget/rerender.tsx [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+// import 'antd/dist/antd.css';
+import PreviewTitle from './components/PreviewTitle';
+import PreviewContent from './components/PreviewContent';
+import Note from './components/Note';
+
+const AppSpace = ({listContent = [], titleWidget = ''}: {listContent: ModalAppSpace[]; titleWidget: string}) => {  
+  const [listAppSpace, setListAppSpace] = useState(listContent);
+  const [titleAppSpace, setTitleAppSpace] = useState(titleWidget); 
+  
+  useEffect(() => {
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/rerender.tsx]: Re-run eslint

File: src/renderer/Widget/AppSpaceListWidget/service.ts [hunk 41]
```diff
@@ -0,0 +1,64 @@
+import {PRESET_ICON_APP} from './constant';
+
+const getAppInfo: ({listAppSpace}: {listAppSpace: ModalAppSpace[]}) => Promise<ModalAppSpace[]> = async ({
+  listAppSpace,
+}: {
+  listAppSpace: ModalAppSpace[];
+}) => {
+  let newListAppSpace = listAppSpace.slice();
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/service.ts]: Re-run eslint

File: src/renderer/Widget/AppSpaceListWidget/service.ts [hunk 42]
```diff
@@ -0,0 +1,64 @@
+import {PRESET_ICON_APP} from './constant';
+
+const getAppInfo: ({listAppSpace}: {listAppSpace: ModalAppSpace[]}) => Promise<ModalAppSpace[]> = async ({
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/service.ts]: Rename `ModalAppSpace` for clear meaning

File: src/renderer/Widget/AppSpaceListWidget/type.d.ts [hunk 43]
```diff
@@ -0,0 +1,18 @@
+declare type AppSpaceWidgetProps = {
+  showSettingInit?: boolean;
+  listContent: ModalAppSpace[];
+  titleWidget: string;
+  isSave:boolean
```
Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/type.d.ts]: Re-run eslint

Comment (phongnhat19) [inline src/renderer/Widget/AppSpaceListWidget/type.d.ts]: `isSave` property is not used

File: src/renderer/Widget/constant.ts [hunk 44]
```diff
@@ -5,5 +5,7 @@ export const CONTENT_TYPE = {
   WEATHER: 'Weather',
   GAROON_NOTIFY: 'GNotification',
   DEFAULT: 'DefaultPortal',
-  EMPTY: 'Empty'
+  EMPTY: 'Empty',
+  APP_SPACE:"App/Space"
```
Comment (phongnhat19) [inline src/renderer/Widget/constant.ts]: Re-run eslint

File: src/renderer/views/PortalBuilder/index.tsx [hunk 45]
```diff
@@ -65,6 +65,10 @@ const PortalBuilder = () => {
     {
       icon: <CloudOutlined />,
       name: CONTENT_TYPE.WEATHER
+    },
```
Comment (phongnhat19) [inline src/renderer/views/PortalBuilder/index.tsx]: Re-run eslint

File: src/renderer/views/PortalBuilder/index.tsx [hunk 46]
```diff
@@ -125,7 +129,7 @@ const PortalBuilder = () => {
         <div className="widget-list-container">
           <WidgetList containers={widgetList} />
         </div>
-      </div>
+      </div>
```
Comment (phongnhat19) [inline src/renderer/views/PortalBuilder/index.tsx]: Re-run eslint

**Review by phongnhat19 — CHANGES_REQUESTED** [review]

File: src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx (hunk 16 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/AppSpaceModel.tsx]: I think need React.Fragment, consolog will show error due map need a key key unique.

File: src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx (hunk 34 above)
Comment (Pham-Gia-Huong) [reply src/renderer/Widget/AppSpaceListWidget/components/PreviewContent.tsx]: I think need React.Fragment, consolog will show error due map need a key key unique.

**Review by phongnhat19 — APPROVED** [review]

File: src/renderer/Layout/TabsLayout/style.css [hunk 47]
```diff
@@ -31,7 +31,7 @@
 }
 
 .portal-tabs-layout .kuc-tabs-tabs {
-  min-width: 100%;
+  min-width: 98%;
```
Comment (phongnhat19) [inline src/renderer/Layout/TabsLayout/style.css]: Should not change `Tabs` css

**Review by phongnhat19 — CHANGES_REQUESTED** [review]

**Pull request merged**